openforcefield/utils/toolkits.py

```diff
@@ -1065,9 +1065,16 @@ def from_smiles(self, smiles, hydrogens_are_explicit=False, allow_undefined_ster
         if not (hydrogens_are_explicit):
             result = oechem.OEAddExplicitHydrogens(oemol)
             if result == False:
-                raise Exception(
+                raise ValueError(
                     "Addition of explicit hydrogens failed in from_openeye")
-        # TODO: Add allow_undefined_stereo to this function, and pass to from_openeye?
+        elif hydrogens_are_explicit:
+            if oechem.OEHasImplicitHydrogens(oemol):
```

Comment:
I think you can substitute this nested ifs with a single `elif oechem.OEHasImplicitHydrogens(oemol):`

Reply:
I've met this suggestion half way -- I combined the nested `if` with the `elif`, just to keep the logic clear for future people who work on this file.

Reply:
👍